darts/models/forecasting/forecasting_model.py

```diff
@@ -444,105 +696,202 @@ def historical_forecasts(
+                - (
+                    self.training_sample_time_index_length
+                    if retrain
+                    else self.predict_sample_time_index_length
+                )
+                * series_.freq
+            )
+
+            if historical_forecasts_time_index is None:
+                raise_log(
+                    ValueError(
+                        "Given the provided model, series and covariates, there is no timestamps "
+                        " where we can make a prediction or train the model. "
+                        "Please check the time indexes of the series and covariates."
+                    ),
+                    logger,
                 )
 
-            forecast = self._predict_wrapper(
-                n=forecast_horizon,
-                series=train,
-                past_covariates=past_covariates,
-                future_covariates=future_covariates,
-                num_samples=num_samples,
+            # prepare the start parameter -> pd.Timestamp
+            if start is not None:
+
+                historical_forecasts_time_index = drop_before_index(
+                    historical_forecasts_time_index,
+                    series_.get_timestamp_at_point(start),
+                )
+
+            else:
+                if (retrain is not False) or (not self._fit_called):
+
+                    if train_length:
+                        historical_forecasts_time_index = drop_before_index(
+                            historical_forecasts_time_index,
+                            historical_forecasts_time_index[0]
+                            + max(
+                                (train_length - self.training_sample_time_index_length),
+                                0,
+                            )
+                            * series_.freq,
+                        )
+                    # if not we start training right away, but with 2 minimum points, so we start
+                    # 1 time step after the first trainable point.
+                    # (sklearn check that there are at least 2 points in the training set and it seems
+                    # rather reasonable)
+                    else:
+                        historical_forecasts_time_index = drop_before_index(
+                            historical_forecasts_time_index,
+                            historical_forecasts_time_index[0] + 1 * series_.freq,
+                        )
+
+                    # if retrain is False, fit hasn't been called yet and train_length None,
+                    # it means that the entire backtesting will be based on a set of two training samples
+                    # at the first step, so we warn the user.
+                    if (
+                        (not self._fit_called)
+                        and (retrain is False)
+                        and (not train_length)
+                    ):
+                        logger.warning(
```

Comment:
I wonder whether we should not rather raise an exception in this case. It sounds like almost always a bad (and unexpected) behavior to have a single training done on 2 samples.

Reply:
yes, we might want to do that indeed. More generally, probably not allowing a training under a certain threshold of sample numbers? I fixed 2, but it might be more like a percentage of the total length of the predictable forecast timeindex?

Reply:
Any other threshold except the trivial one (as here) is risky IMO... how would we choose the percentage?